containers/images/core/Dockerfile

```diff
@@ -0,0 +1,30 @@
+FROM centos:7
+
+ENV RAILS_ENV="production"
+ENV FOREMAN_APIPIE_LANGS="en"
+ARG REPO="theforeman/foreman"
+ARG BRANCH="develop"
+ARG TEMP_PACKAGES="curl postgresql-devel gcc-c++ make rh-nodejs8-npm libxml2-devel libcurl-devel qpid-cpp-client-devel rh-ruby24-ruby-devel"
+ARG BUNDLER_SKIPPED_GROUPS="mysql:test:mysql2:development:sqlite:jenkins:openid:libvirt:journald"
+
+WORKDIR /usr/share/foreman
+
+# Downloading foreman source code, using github archive option as its faster
+RUN curl -s https://codeload.github.com/${REPO}/tar.gz/${BRANCH} |tar xzf - --strip-components=1
+
+# Deploy DB configuration, defaulting to postgresql
+ADD container-assets/database.yml config/database.yml
+
+# Plugin configuration section
+# if you want additional plugins please edit the file below
+ADD container-assets/Gemfile.local.rb bundler.d/Gemfile.local.rb
+
+# Foreman setting
+ADD container-assets/settings.yaml config/settings.yaml
+
+# Build script
+ADD container-assets/setup tmp/setup
+ADD container-assets/build tmp/build
+RUN tmp/setup
```

Comment:
This implies a single layer for all functionality contained within this. Which is one aspect I thought we in theory wanted to avoid? I strayed away from continuing to use Ansible in part due to this. If it does not matter much, I would lean towards us going back to using Ansible given the clean syntax. Perhaps, as @carbonin pointed out, ansible-runner might make that easier on us.

Reply:
Agree, was the intention for other images to be built by the user adding scripts and building this one rather than using dependent images?

I think making this a base image and then allowing users to build in plugins by using something like `FROM foreman-base:latest` might be a better architecture.

Reply:
TBH: this was just a optimization phase to run it all on a single RUN vs not, numbering the files just makes it clear which one runs first. I have no problem to replace the setup script with hard coded steps, if i add them as distinct RUN steps in the docker file they add about 500MB to the image...